src/ExportMap.js

```diff
@@ -514,30 +514,50 @@ ExportMap.parse = function (path, content, context) {
 
     // This doesn't declare anything, but changes what's being exported.
     if (n.type === 'TSExportAssignment') {
-      const moduleDecls = ast.body.filter((bodyNode) =>
-        bodyNode.type === 'TSModuleDeclaration' && bodyNode.id.name === n.expression.name
+      const exportedName = n.expression.name
+      const declTypes = [
+        'VariableDeclaration',
+        'ClassDeclaration',
+        'TSDeclareFunction',
+        'TSEnumDeclaration',
+        'TSTypeAliasDeclaration',
+        'TSInterfaceDeclaration',
+        'TSAbstractClassDeclaration',
+        'TSModuleDeclaration',
+      ]
+      const exportedDecls = ast.body.filter(({ type, id, declarations }) => 
+        declTypes.includes(type) && 
+        (id && id.name === exportedName || declarations.find(d => d.id.name === exportedName))
       )
-      moduleDecls.forEach((moduleDecl) => {
-        if (moduleDecl && moduleDecl.body && moduleDecl.body.body) {
-          moduleDecl.body.body.forEach((moduleBlockNode) => {
+      if (exportedDecls.length === 0) {
+        // Export is not referencing any local declaration, must be re-exporting
+        m.namespace.set('default', captureDoc(source, docStyleParsers, n))
```

Comment:
I'm not sure about `n` here. Would it be worth looking through the `m.imports` to see if any matches the exported name? We don't do that on [regular exports](https://github.com/benmosher/eslint-plugin-import/pull/1528/files#diff-e991b36c6628220049668b5e2d2c43d8R501). E.g.:

```
import { test } from './test';
export { foo }; // all good...
```